lib/internal/encoding.js

```diff
@@ -380,13 +386,12 @@ function makeTextDecoderICU() {
 
     decode(input = empty, options = {}) {
       validateDecoder(this);
-      if (isArrayBuffer(input)) {
+      validateArgumentBuffer(input, 'input');
+
+      if (isArrayBuffer(input))
```

Comment:
This results in a second `isArrayBuffer` check and is definitely not desirable. Please keep that check as it was before (as in: remove `validateArgumentBuffer` again).

Reply:
ok.